presto-main-base/src/main/java/com/facebook/presto/sql/planner/GroupedExecutionTagger.java

```diff
@@ -161,15 +158,8 @@ public GroupedExecutionTagger.GroupedExecutionProperties visitMergeJoin(MergeJoi
                     left.totalLifespans,
                     left.recoveryEligible && right.recoveryEligible);
         }
-        throw new PrestoException(
-                INVALID_PLAN_ERROR,
-                format("When grouped execution can't be enabled, merge join plan is not valid." +
-                                "%s is currently set to %s; left node grouped execution capable is %s and " +
-                                "right node grouped execution capable is %s.",
-                        GROUPED_EXECUTION,
-                        groupedExecutionEnabled,
-                        left.currentNodeCapable,
-                        right.currentNodeCapable));
+        // TODO: This will break the other use case for merge join operating on sorted tables, which requires grouped execution for correctness.
+        return GroupedExecutionTagger.GroupedExecutionProperties.notCapable();
```

Comment:
What's the difference here?

Reply:
`GroupedExecutionTagger` is a sanity check for group execution. Previously, merge join has to be executed with group execution and if its inputs are not with group execution, it throws an error here.

Now with SMJ, group execution is no longer a required for merge join, so we need to relax the check but marking this subplan doesn't have group execution (so the upstream nodes know). This might break merge join without sorting because it still needs this sanity check.